File: backend/app/schemas/pcp.py
"""
Schemas: PCP (Planning & Control)
"""
from pydantic import BaseModel
from datetime import datetime


class PcpMessageCreate(BaseModel):
    message: str
    priority: int = 0
    type: str = "info"
    target_machine: str | None = None
    created_by: str | None = None
    expires_at: datetime | None = None


class PcpMessageRead(BaseModel):
    id: int
    message: str
    priority: int
    type: str
    target_machine: str | None = None
    is_active: bool
    created_by: str | None = None
    expires_at: datetime | None = None
    created_at: datetime | None = None

    model_config = {"from_attributes": True}

File: synchro-mes/frontend/src/types/pcp.ts
// ── PCP ──────────────────────────────────────────────────────

export interface PcpMessage {
  id: number;
  message: string;
  priority: number;
  type: string;
  target_machine: string | null;
  is_active: boolean;
  created_by: string | null;
  expires_at: string | null;
  created_at: string | null;
}

export interface PcpMessageCreate {
  message: string;
  priority?: number;
  type?: string;
  target_machine?: string;
  created_by?: string;
  expires_at?: string;
}
